pom.xml

```diff
@@ -107,11 +108,11 @@
     </profiles>
 
     <properties>
-        <axon.version>4.0.1</axon.version>
+        <axon.version>4.2</axon.version>
```

Comment:
We'd might want to change this back to 4.0, so that we can justify a Kafka Extension 4.0 release for sake of compatibility with older versions. This is more so a statement than an immediate change request from my part, as this statement is, in my honest opinion, still open for debate.

Reply:
My suggestion is to release a 4.0.1, create a 4.2 branch and immediately release 4.2

I just want to have the latest version supported, it indicates that the extension is up to the latest released version. Until you are not planing to use extension versioning as a driving force for the community to update to latest Axon version, I assume we should create as many "matching" versions as possible (understanding the Axon versioning politics, I think every minor is worth a extension release) and putting somewhere a note that no back-porting of features/bugs is intended in extensions.

So I'll patch the version back to 4.0.1 for this PR.

Reply:
Thanks for that @zambrovski! In practicality it should be fine to go immediately for a 4.2 release of Kafka, as it's been a release candidate for quite sometime now. However, we already had a Kafka module in Axon 3.x, so I am a bit torn between both approaches at this stage.
Moreover, I feel it is nicer to the community to show we've had a Kafka Module/Extension tailing along with the main versioning scheme (which is like you assumed, having Framework/Server take the extensions along).